beautiful-racket-lib/br/private/generate-literals.rkt

```diff
@@ -9,13 +9,46 @@
          (or (not (regexp-match #rx"[A-Z]" pat-str)) ; either doesn't contain at least one uppercase letter ...
              (not (equal? (string-upcase pat-str) pat-str)))))) ;...  or doesn't contain all uppercase letters
 
+(define (wildcard-identifier? pat-datum)
+  (and (symbol? pat-datum)
+       (not (literal-identifier? pat-datum))
+       (not (memq pat-datum '(... _)))))
+
 ;; generate literals for any symbols that are not ... or _ and not IN_CAPS
 (define (generate-literals pats)
   (for*/list ([pat-arg (in-list (syntax-flatten pats))]
               [pat-datum (in-value (syntax->datum pat-arg))]
               #:when (literal-identifier? pat-datum))
     pat-arg))
 
+(define (all-...-follow-wildcards pats)
+  (define prev-datum (box #f))
+  (and
+    (for*/and ([pat-arg (in-list (syntax-flatten pats))]
+               [pat-datum (in-value (syntax->datum pat-arg))])
+      ;; OK if there's no previous datum,
```

Comment:
Oh no, I can’t mess with perfection. 

I was just trying to understand what `box` is for. IIUC it’s a way of saving the costs of allocation & GC across function calls by passing what’s essentially a reference to a value rather than the value itself. But, if that’s so, then I don’t understand the benefit of putting a Boolean or symbol in a box, since they’re so tiny to begin with.